### DISCUSSION TITLE
Training universal pwgan background noise and bad output

### DISCUSSION THREAD

**Discussion by JRMeyer** [coqui-ai/TTS#243] [Archived]
<USER>>>> georroussos<br/> <DATE> [July 23, 2020, 9:40am]<br/> <br/> I have been trying to train a universal ParallelWaveGAN for a long while<br/> now but I have not really had any success. I first started with a<br/> private single speaker and it worked very well with no background noise,<br/> so I thought I would try my luck with LibriTTS. My understanding was<br/> that the network would be exposed to many, many more waveforms, so it<br/> would be able to produce speech for any speaker. The models I have been<br/> training have all been outputting sound with artifacts and background<br/> noise, like a static 'zzzz'. I thought I could introduce some more<br/> private speaker to LibriTTS, so I did (sum was Libri-360 and 200 more<br/> speakers). The results did not really change. At best, they sound<br/> muffled. I would never expect WaveRNN quality, but I would at least<br/> expect single speaker quality. I also do not know what to make of my<br/> tensorboard logs, but I think they do look pretty erratic.<br/> <br/> format: 'hdf5'<br/> audio:<br/> clip_norm: true<br/> do_trim_silence: false<br/> frame_length_ms: 50<br/> frame_shift_ms: 12.5<br/> max_norm: 4<br/> hop_length: 275<br/> win_length: 1100<br/> mel_fmax: 8000.0<br/> mel_fmin: 0.0<br/> min_level_db: -100<br/> num_freq: 1025<br/> num_mels: 80<br/> preemphasis: 0.98<br/> ref_level_db: 20<br/> sample_rate: 22050<br/> signal_norm: true<br/> sound_norm: false<br/> symmetric_norm: true<br/> trim_db: 20<br/> <br/> generator_params:<br/> in_channels: 1 # Number of input channels.<br/> out_channels: 1 # Number of output channels.<br/> kernel_size: 3 # Kernel size of dilated convolution.<br/> layers: 30 # Number of residual block layers.<br/> stacks: 3 # Number of stacks i.e., dilation cycles.<br/> residual_channels: 64 # Number of channels in residual conv.<br/> gate_channels: 128 # Number of channels in gated conv.<br/> skip_channels: 64 # Number of channels in skip conv.<br/> aux_channels: 80 # Number of channels for auxiliary feature conv.<br/> # Must be the same as num_mels.<br/> aux_context_window: 2 # Context window size for auxiliary feature.<br/> # If set to 2, previous 2 and future 2 frames will be considered.<br/> dropout: 0.0 # Dropout rate. 0.0 means no dropout applied.<br/> use_weight_norm: true # Whether to use weight norm.<br/> # If set to true, it will be applied to all of the conv layers.<br/> upsample_net: 'ConvInUpsampleNetwork' # Upsampling network architecture.<br/> upsample_params: # Upsampling network parameters.<br/> upsample_scales:<br/> - 5<br/> - 5<br/> - 11<br/> <br/> discriminator_params:<br/> in_channels: 1 # Number of input channels.<br/> out_channels: 1 # Number of output channels.<br/> kernel_size: 3 # Number of output channels.<br/> layers: 10 # Number of conv layers.<br/> conv_channels: 64 # Number of chnn layers.<br/> bias: true # Whether to use bias parameter in conv.<br/> use_weight_norm: true # Whether to use weight norm.<br/> # If set to true, it will be applied to all of the conv layers.<br/> nonlinear_activation: 'LeakyReLU' # Nonlinear function after each conv.<br/> nonlinear_activation_params: # Nonlinear function parameters<br/> negative_slope: 0.2 # Alpha in LeakyReLU.<br/> <br/> stft_loss_params:<br/> fft_sizes: [1024, 2048, 512] # List of FFT size for STFT-based loss.<br/> hop_sizes: [120, 240, 50] # List of hop size for STFT-based loss<br/> win_lengths: [600, 1200, 240] # List of window length for STFT-based loss.<br/> window: 'hann_window' # Window function for STFT-based loss<br/> <br/> lambda_adv: 4.0 # Loss balancing coefficient.<br/> <br/> batch_size: 8 # Batch size.<br/> batch_max_steps: 26125 # Length of each audio in batch. Make sure dividable by hop_size.<br/> pin_memory: true # Whether to pin memory in Pytorch DataLoader.<br/> num_workers: 8 # Number of workers in Pytorch DataLoader.<br/> remove_short_samples: true # Whether to remove samples the length of which are less than batch_max_steps.<br/> allow_cache: false # Whether to allow cache in dataset. If true, it requires cpu memory.<br/> <br/> generator_optimizer_params:<br/> lr: 0.0001 # Generator's learning rate.<br/> eps: 1.0e-6 # Generator's epsilon.<br/> weight_decay: 0.0 # Generator's weight decay coefficient.<br/> generator_scheduler_params:<br/> step_size: 200000 # Generator's scheduler step size.<br/> gamma: 0.5 # Generator's scheduler gamma.<br/> # At each step size, lr will be multiplied by this parameter.<br/> generator_grad_norm: 10 # Generator's gradient norm.<br/> discriminator_optimizer_params:<br/> lr: 0.00005 # Discriminator's learning rate.<br/> eps: 1.0e-6 # Discriminator's epsilon.<br/> weight_decay: 0.0 # Discriminator's weight decay coefficient.<br/> discriminator_scheduler_params:<br/> step_size: 200000 # Discriminator's scheduler step size.<br/> gamma: 0.5 # Discriminator's scheduler gamma.<br/> # At each step size, lr will be multiplied by this parameter.<br/> discriminator_grad_norm: 1 # Discriminator's gradient norm.<br/> <br/> discriminator_train_start_steps: 100000 # Number of steps to start to train discriminator.<br/> train_max_steps: 1000000 # Number of training steps.<br/> save_interval_steps: 5000 # Interval steps to save checkpoint.<br/> eval_interval_steps: 1000 # Interval steps to evaluate the network.<br/> log_interval_steps: 100 # Interval steps to record the training log.<br/> <br/> num_save_intermediate_results: 4 # Number of results to be saved as intermediate results.<br/> <br/> I have downsampled the dataset and changed the hop and win sizes to<br/> accomodate with my TTS attributes.<br/> <br/> 15.02.43.filename}[2880×1258 421<br/> <br/> 15.02.49.filename}[2880×1258 551<br/> <br/> [This is an archived TTS discussion thread from discourse.mozilla.org/t/training-universal-pwgan-background-noise-and-bad-output]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> erogol<br/> <DATE> [July 23, 2020, 1:00pm]<br/> <br/> did you change the libriTTS sampling rate since the default is 24khz I<br/> think which is not compatible with your config file?<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> georroussos<br/> <DATE> [July 23, 2020, 1:04pm]<br/> <br/> I did! I downsampled LibriTTS to 22050 to match my TTS.<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> erogol<br/> <DATE> [July 23, 2020, 1:05pm]<br/> <br/> have you listened the samples after down sampling ?<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> erogol<br/> <DATE> [July 23, 2020, 1:05pm]<br/> <br/> Ohh you tried the other PWGAN repo. Maybe you can try the one under TTS<br/> dev branch.<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> georroussos<br/> <DATE> [July 23, 2020, 1:07pm]<br/> <br/> I did that also, they sounded fine. Yes I trained using the initial<br/> PWGAN fork. Is the one under TTS dev different? I can try that one<br/> instead. Does it have anything different?<br/> <br/> [Archived Post]<br/>
